src/htmx.js

```diff
@@ -3102,7 +3102,7 @@ return (function () {
 
             // behavior of anchors w/ empty href is to use the current URL
             if (path == null || path === "") {
-                path = getDocument().location.href;
+                path = getDocument().location.href.split('?')[0];
```

Comment:
As the above comment indicates, this is also applied to anchors with an empty `href` attribute and that changes would be a change in behavior. Not sure if we should fix it here or better move the handling to https://github.com/bigskysoftware/htmx/blob/7274454360aa5ff5f9a5ed1657450b93e9529b89/src/htmx.js#L1372